staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go

```diff
@@ -350,15 +350,15 @@ func isBackendPoolOnSameLB(newBackendPoolID string, existingBackendPools []strin
 	}
 
 	newLBName := matches[1]
-	newLBNameTrimmed := strings.TrimRight(newLBName, InternalLoadBalancerNameSuffix)
+	newLBNameTrimmed := strings.TrimSuffix(newLBName, InternalLoadBalancerNameSuffix)
 	for _, backendPool := range existingBackendPools {
 		matches := backendPoolIDRE.FindStringSubmatch(backendPool)
 		if len(matches) != 2 {
 			return false, "", fmt.Errorf("existing backendPoolID %q is in wrong format", backendPool)
 		}
 
 		lbName := matches[1]
-		if !strings.EqualFold(strings.TrimRight(lbName, InternalLoadBalancerNameSuffix), newLBNameTrimmed) {
+		if !strings.EqualFold(strings.TrimSuffix(lbName, InternalLoadBalancerNameSuffix), newLBNameTrimmed) {
```

Comment:
what's diff between `TrimSuffix` and `TrimRight` func?

Reply:
TrimSuffix is word-based and TrimRight is character-based.
according the work these 2 lines want to do, it just want to delete suffix word in the string, so I think TrimSuffix is suitable.

Reply:
would it be better to split this into a bugfix PR and keep this one focused on static check failures?

Reply:
Sure, I will file a related pr, thanks.